src/chanakya/web/routes.py

```
@@ -175,91 +246,141 @@ async def record():
+            with tempfile.NamedTemporaryFile(delete=False, suffix=".wav") as tts_f:
+                tts_f.write(tts_audio_bytes)
+                audio_file_path = tts_f.name
             if not audio_file_path or not os.path.exists(audio_file_path):
                 return jsonify({"error": "TTS audio file not found or not created."}), 500
-            with open(audio_file_path, "rb") as f_audio: audio_bytes = f_audio.read()
-            audio_base64 = base64.b64encode(audio_bytes).decode('utf-8')
+            with open(audio_file_path, "rb") as f_audio:
+                audio_bytes = f_audio.read()
+            audio_base64 = base64.b64encode(audio_bytes).decode("utf-8")
             return jsonify({"audio_data_url": f"data:audio/wav;base64,{audio_base64}"})
         except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)